cpp/src/prims/detail/nbr_intersection.cuh

```diff
@@ -436,33 +524,63 @@ struct strided_accumulate_t {
   }
 };
 
-template <typename vertex_t>
+template <typename vertex_t, typename EdgeProperty_t>
 struct gatherv_indices_t {
   size_t output_size{};
   int minor_comm_size{};
 
   raft::device_span<size_t const> gathered_intersection_offsets{};
   raft::device_span<vertex_t const> gathered_intersection_indices{};
   raft::device_span<size_t const> combined_nbr_intersection_offsets{};
-
   raft::device_span<vertex_t> combined_nbr_intersection_indices{};
 
+  std::optional<raft::device_span<EdgeProperty_t const>> gathered_nbr_intersection_properties0{
+    std::nullopt};
+  std::optional<raft::device_span<EdgeProperty_t const>> gathered_nbr_intersection_properties1{
+    std::nullopt};
+
+  std::optional<raft::device_span<EdgeProperty_t>> combined_nbr_intersection_properties0{
+    std::nullopt};
+  std::optional<raft::device_span<EdgeProperty_t>> combined_nbr_intersection_properties1{
+    std::nullopt};
```

Comment:
Did this work? You can't pass std::optional to device, use thrust::optional for device functors.